.grit/patterns/js/react_hooks.grit

```diff
@@ -44,19 +44,16 @@ private pattern handle_one_statement($class_name, $statements, $states_statement
             or {
                 and {
                     $name <: js"constructor",
-                    $body <: maybe contains bubble($constructor_statements) {
-                        or {
-                            lexical_declaration($declarations) as $decl where $declarations <:
-                                variable_declarator($name, $value) where {
-                                    $name <: not contains js"this.state",
-                                    $value <: not contains js"this.state",
-                                    $constructor_statements += $decl
-                                },
-                            expression_statement() as $decl where {
-                                $decl <: not contains `super($props)`,
-                                $decl <: not contains `$_.bind($_)`,
-                                $constructor_statements += $decl,
-                            },
+                    $body <: maybe contains or {
```

Comment:
I added the `maybe` here in case the original constructor will not contain anything from the match. 

Test here: https://github.com/getgrit/stdlib/pull/71/files#diff-5a88cdcb5dc8df71e57e7aa26d8901edc02231fc9d1ba7c74bfc05c29a3c5f44R1301-R1325